streamlit_drawable_canvas/frontend/src/lib/line.ts

```diff
@@ -58,6 +61,10 @@ class LineTool extends FabricTool {
 
   onMouseUp(o: any) {
     this.isMouseDown = false
+    let canvas = this._canvas
+    if (this.currentLine.width === 0 && this.currentLine.height === 0) {
+      canvas.remove(this.currentLine)
+    }
```

Comment:
According to my understanding, the `line` only has its "head and tail" after the mouse released. So my hack is to remove lines which has zero width and height after it has been added to canvas.